draco/analysis/radiometer_test.py

```diff
@@ -0,0 +1,238 @@
+        ind_XX = [] #Index of polarization within the selection
+        ind_YY = []
+        auto_pol_pos = 0 #Counter for getting the index
+        for ii in range(len(data.index_map["prod"][:])):
+            feedi, feedj = self.telescope.uniquepairs[ii]
+            pol_0        = self.telescope.polarisation[feedi]
+            pol_1        = self.telescope.polarisation[feedj]
+            if feedi==feedj:
+                auto_stack_id.append(ii)
+                if pol_0=='X' and pol_1=='X':
+                    ind_XX.append(auto_pol_pos)
+                elif pol_0=='Y' and pol_1=='Y':
+                    ind_YY.append(auto_pol_pos)
+                auto_pol_pos +=1
+
+        bflag = (data.weight.data[:,auto_stack_id,:] == 0.0)
+        bvis  = data.vis.data[:,auto_stack_id:,]
+	bfrac = data.flags['frac_lost'].data
+	binp  = data.flags['inputs']
+        time_auto = data.time
+
+	#Derive weights for each cylinder, based on the number of active feeds
+	live_feed_cyl_XX = np.zeros((ncyl, len(time_auto)))
+	live_feed_cyl_YY = np.zeros((ncyl, len(time_auto)))
+
+	for i in range(ncyl):
+	    live_feed_cyl_XX[i] = live_feeds(chr(i+ord('A')),'XX',binp)
+	    live_feed_cyl_YY[i] = live_feeds(chr(i+ord('A')),'YY',binp)
+
+	#Average autos across cylinders
+	vis_avg_XX = np.zeros((nfreq,len(time_auto)), dtype='complex64')
+	vis_avg_YY = np.zeros((nfreq,len(time_auto)), dtype='complex64')
+
+	#Averaging for pol XX (multiplying visibilities pairwise between different cylinders)
+	for i in ind_XX:
+	    for j in ind_XX:
+		if i >= j: #Avoid double counting
+		    continue 
+		
+                n_feeds_cyl_XX = live_feeds(chr(i+ord('A')),'XX',binp)*live_feeds(chr(j+ord('A')),'XX',binp) #Total number of feeds in this product
+		vis_avg_XX +=  bvis[:,i,:]*bvis[:,j,:] * (n_feeds_cyl_XX/feed_qnt**2) # Product of visibilities weighted by fraction of non-flagged feeds
+		
+                Nbl_XX += n_feeds_cyl_XX #Counter for total number of feeds averaged
+                N_XX += n_feeds_cyl_XX/feed_qnt**2 #Effective number of total averages, max: 6 
+		
+	#Averaging for pol YY
+	for i in ind_YY:
+	    for j in ind_YY:
+		if i >= j: 
+		    continue 
+		
+                n_feeds_cyl_YY = live_feeds(chr(i+ord('A')-len(ind_XX)),'YY',binp)*live_feeds(chr(j+ord('A')-len(ind_XX)),'YY',binp)
+		vis_avg_YY +=  bvis[:,i,:] * bvis[:,j,:] * n_feeds_cyl_YY/feed_qnt**2
+		
+                Nbl_YY += n_feeds_cyl_YY
+		N_YY += n_feeds_cyl_YY/feed_qnt**2
+
+	#Get the quantities for normalization: 
+	#integration time, number of baselines, fraction of packets lost and frequency resolution
+	tint = np.median(np.diff(time_auto)) #there is a slight different in integrations at micro seconds, ignoring it
+        spw  = np.median([i[1] for i in auto_data.index_map['freq']]) #assuming same spectral bandwidth across the frequency
+
+	n_b_tau_XX = spw * tint * Nbl_XX * (1-bfrac)
+	n_b_tau_YY = spw * tint * Nbl_YY * (1-bfrac)
+
```

Comment:
If you did as suggested above, then this would become:
```
nint = spw * tint * (1.0 - bfrac[:, np.newaxis, :])
radiometer = np.sqrt(radiometer * tools.invert_no_zero(nint * radiometer_counter**2 ))
```
which could be saved directly to the container.